runtime/compiler/ilgen/Walker.cpp

```diff
@@ -6327,7 +6388,9 @@ static TR::SymbolReference * createLoadFieldSymRef(TR::Compilation * comp, TR_Op
          }
       }
 
-   TR_ASSERT_FATAL(false, "Did not find the matching fieldname %s", fieldname);
+   if (failIfNotFound)
+      TR_ASSERT_FATAL(false, "Did not find the matching fieldname %s", fieldname);
```

Comment:
This is a real nit, but maybe this should just be

```
TR_ASSERT_FATAL(!failIfNotFound, "Did not find the matching fieldname %s", fieldname);
```

Or maybe you could avoid the double negation by making the parameter `bool passIfNotFound = false` and make the assertion into

```
TR_ASSERT_FATAL(passIfNotFound, "Did not find the matching fieldname %s", fieldname);
```

Reply:
😮‍💨 I wished I had thought of that! I like this a lot more

Reply:
Or maybe `nullIfNotFound` might be better